### PR TITLE
Avoid conditional compilation on whether tracing is enabled

### PR DESCRIPTION
**Pull request by vector-of-bool**
Macros that discard their arguments under certain configurations can lead to missed refactors and unknowingly breaking code. This changeset changes conditional tracing to use C-language-level conditions rather than preprocessor conditions.

### CODE REVIEW

**Review by eramongodb — APPROVED** [review]
I assume these changes suggest the runtime impact of disabled tracing utilities is not significant enough to justify their conditional compile-time compilation and corresponding maintenance burden. It seems like the compilers are also gracious enough not to warn about the always true/false conditions predecated on `MONGOC_TRACE_ENABLED`. LGTM.

File: src/libmongoc/src/mongoc/mongoc-log.c [hunk 1]
```diff
@@ -85,10 +103,8 @@ mongoc_log (mongoc_log_level_t log_level,
    bson_once (&once, &_mongoc_ensure_mutex_once);
 
    stop_logging = !gLogFunc;
```
Comment (chardan) [inline src/libmongoc/src/mongoc/mongoc-log.c:105]: Should this check be combined into _mongoc_log_trace_is_enabled()?

**Review by chardan — APPROVED** [review]
+1, LGTM!

**Review by kevinAlbs — APPROVED** [review]

File: src/libmongoc/src/mongoc/mongoc-log.c (hunk 1 above)
Comment (vector-of-bool) [reply src/libmongoc/src/mongoc/mongoc-log.c:105]: Unsure. I left the logic as close to the original as possible.

File: src/libmongoc/src/mongoc/mongoc-config.h.in [hunk 2]
```diff
@@ -353,13 +365,18 @@
 /*
  * Set if tracing is enabled. Logs things like network communication and
  * entry/exit of certain functions.
- *
  */
-#define MONGOC_TRACE @MONGOC_TRACE@
+#cmakedefine01 MONGOC_TRACE
```
Comment (jmikola) [inline src/libmongoc/src/mongoc/mongoc-config.h.in:369]: This unexpectedly broke PHPC's build system (autotools). I've reported [CDRIVER-4247](https://jira.mongodb.org/browse/CDRIVER-4247) to request a more portable approach.

Comment (eramongodb) [reply src/libmongoc/src/mongoc/mongoc-config.h.in:369]: Resolved in https://github.com/mongodb/mongo-c-driver/pull/917.

**Pull request merged**